dashboard/app/models/sections/section.rb

```diff
@@ -248,17 +248,7 @@ def update_student_sharing(sharing_disabled)
   def ensure_owner_is_active_instructor
     return if user.blank?
 
-    si = SectionInstructor.with_deleted.find_by(instructor: user, section_id: id)
-    if si.blank?
-      SectionInstructor.create!(section_id: id, instructor: user, status: :active)
-    elsif si.deleted?
-      si.restore
-      si.status = :active
-      si.save!
-    elsif si.status != 'active'
-      si.status = :active
-      si.save!
-    end
+    add_instructor(user)
```

Comment:
This hook was meant to ensure that we never get into a state where a section would be left without an instructor, as the section owner will always be an active instructor.

`add_instructor` does not provide a strong guarantee that it will actually add an instructor (if the user is a student), so I worry this will introduce the possibility of introducing orphaned sections. I suggest not relaxing this guarantee here.